### PR TITLE
testcase for /dev/crypto

### PR DESCRIPTION
**Pull request by anjiahao1**
## Summary
some testcase for /dev/crypto,it code form openbsd
## Testing
run it & pass ci

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: testing/crypto/hmac.c [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+#include <crypto/cryptodev.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <crypto/md5.h>
+#include <crypto/sha1.h>
+#include <crypto/sha2.h>
+
+struct tb
+{
+  char *key;
```
Comment (pkarashchenko) [inline testing/crypto/hmac.c:34]: Are this tests targeted to run on PC or on the MCU? If it is on MCU, then we need to add `FAR` to pointers

**Review by hartmannathan — CHANGES_REQUESTED** [review]
Need to add the two BSD-licensed files to LICENSE.

**Comment by anjiahao1** [comment]
> 

ok done

**Comment by anjiahao1** [comment]
> Need to add the two BSD-licensed files to LICENSE.

ok done

**Review by hartmannathan — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**